client/src/main/java/org/eclipse/hono/client/impl/HonoClientImpl.java

```diff
@@ -169,6 +169,7 @@ public HonoClient connect(
                     conAttempt -> {
                         connecting.compareAndSet(true, false);
                         if (conAttempt.failed()) {
+                            LOG.info("Failed to establish connection", conAttempt.cause());
```

Comment:
there is no need to log this here (again). It is already logged in `org.eclipse.hono.connection.ConnectionFactoryImpl`
In fact, logging this information is kind of a two-sided sword. On the one hand it is useful to have the information if you actually expect the connection to get established. However, during runtime, it is totally expected that a connection breaks down, e.g. because the peer service is stopped/restarted/whatever. The client accounts for this by means of automatically trying to re-connect until the connection is established. So, under *normal* conditions, logging this at INFO would simply flood the log with meaningless information. In the `ConnectionFactoryImpl` class we therefore log the information at DEBUG level.